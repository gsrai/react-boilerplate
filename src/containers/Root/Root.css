html, body {
  height: 100%;
  margin: 0;
}

#application-root {
  height: 100%;
  color: blue;
}