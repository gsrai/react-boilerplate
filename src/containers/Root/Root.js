import React from 'react'
import { Link, Route } from 'react-router-dom'
import { hot } from 'react-hot-loader'
import './Root.css'

import Counter from '../../components/Counter/Counter'
import Greeter from '../../components/Greeter/Greeter'
import RepoList from '../RepoList/RepoList'

const Root = (props) => (
  <div id='application-root'>
    <h1><Greeter greeting={'Bonjour!'} />, Using React {React.version}</h1>
    <Counter />
    <nav>
      <Link to={'/repolist'}>Repo List</Link>
    </nav>
    <div>
      <Route path={'/repolist'} component={RepoList} />
    </div>
  </div>
)

export default hot(module)(Root)
