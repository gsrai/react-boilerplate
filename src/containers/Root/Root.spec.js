import { shallow } from 'enzyme'
import React from 'react'
import Root from './Root'

describe('Component Rendering', () => {
  test('component should render correctly', () => {
    const wrapper = shallow(<Root />)
    expect(wrapper).toMatchSnapshot()
  })
})
