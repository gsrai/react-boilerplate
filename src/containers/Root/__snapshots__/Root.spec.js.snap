// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Rendering component should render correctly 1`] = `
<div
  id="application-root"
>
  <h1>
    <Greeter
      greeting="Bonjour!"
    />
    , Using React 
    16.4.2
  </h1>
  <Counter />
  <nav>
    <Link
      replace={false}
      to="/repolist"
    >
      Repo List
    </Link>
  </nav>
  <div>
    <Route
      component={[Function]}
      path="/repolist"
    />
  </div>
</div>
`;
